File: ruby/domain-reputation-sample.rb
require 'erb'
require 'json'
require 'net/https'
require 'uri'
require 'yaml'

api_key = 'Your Reputation Scoring API key'
domain = 'threatintelligenceplatform.com'
mode = 'full'

api_url = 'https://api.threatintelligenceplatform.com/v1/reputation'

url = api_url +
      '?apiKey=' + ERB::Util.url_encode(api_key) +
      '&domainName=' + ERB::Util.url_encode(domain) +
      '&mode=' + ERB::Util.url_encode(mode)

begin
  buffer = Net::HTTP.get(URI.parse(url))
  json = JSON.parse(buffer)
  puts json.to_yaml
rescue Errno::ECONNRESET, Errno::EINVAL, EOFError, JSON::ParserError,
       Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError, Net::ProtocolError,
       OpenSSL::SSL::SSLError, Timeout::Error, URI::InvalidURIError => e
  puts e
end

File: php/domain-reputation-sample.php
<?php

$apiUrl = 'https://api.threatintelligenceplatform.com/v1/reputation';

$params = array(
    'apiKey' => 'Your domain reputation API key',
    'domainName' => 'threatintelligenceplatform.com',
    'mode' => 'fast'
);

$url = $apiUrl . '?' . http_build_query($params, '', '&');

print($url . PHP_EOL . PHP_EOL);

$response = file_get_contents($url);

$res = json_decode($response);

if (! $res)
    return;

$score = $res->reputationScore;
if (! $score)
    return;

echo 'Reputation score: ' . $score . PHP_EOL;